### PR TITLE
feat: introduce new module token-validation-spring-test for JUnit tests and local development

### PR DESCRIPTION
**Pull request by tommytroen**
* Make use of [mock-oauth2-server](https://github.com/navikt/mock-oauth2-server) for tests and local development
* Spring auto-configuration supporting unlimited amount of issuers
* Deprecate token-validation-test-support module

### CODE REVIEW

**Comment by tommytroen** [comment]
fix #29

**Comment by tommytroen** [comment]
@janolaveide added you as a reviewer due to the Spring nature of this change 😄 Needed to have the mock server started before the application context of the app using token-support was finished. Any feedback is welcome!

**Review by ybelMekk — APPROVED** [review]
LGTM. Im not a spring guru. but this looks good to me :D

File: token-validation-spring-demo/src/test/java/no/nav/security/token/support/demo/spring/LocalSecurityConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import no.nav.security.token.support.spring.test.EnableMockOAuth2Server;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+@Profile("local")
+@EnableMockOAuth2Server
+public class LocalSecurityConfiguration {
+
```
Comment (jksolbakken) [inline token-validation-spring-demo/src/test/java/no/nav/security/token/support/demo/spring/LocalSecurityConfiguration.java:11]: Why an empty class?

**Review by jksolbakken — APPROVED** [review]

File: token-validation-spring-demo/src/test/java/no/nav/security/token/support/demo/spring/LocalSecurityConfiguration.java (hunk 1 above)
Comment (tommytroen) [reply token-validation-spring-demo/src/test/java/no/nav/security/token/support/demo/spring/LocalSecurityConfiguration.java:11]: In a normal app this class would contain some configuration, created here to show usage of `@EnableMockOAuth2Server` and @Profile annotations.

**Pull request merged**